components/operator/internal/controllers/stack/stack_reconciler.go

```diff
@@ -114,6 +114,33 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}, nil
 	}
 
+	conf := &stackv1beta3.Configuration{}
+	if err := r.client.Get(ctx, types.NamespacedName{
+		Namespace: "",
+		Name:      stack.Spec.Seed,
+	}, conf); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	stackFinalizer := NewStackFinalizer(
+		ctx,
+		r.client,
+		log,
+		r.configuration,
+		stack,
+		conf,
+		&stackv1beta3.Versions{},
+	)
+
+	res, err := stackFinalizer.HandleFinalizer(ctx, log, stack, conf, req)
```

Comment:
Your finalizer should indicate the object was deleted to allow this function to stop the process.

Reply:
There is watch/polling. As the operator has to be idempotent, I just try to get the ressource and see if i get a 404

Reply:
Now just a boolean when the process suceed

Reply:
Else if there is an error, we requeue